lemmy_apub/src/objects/mod.rs

```diff
@@ -16,6 +18,46 @@ pub(crate) mod post;
 pub(crate) mod private_message;
 pub(crate) mod user;
 
+/// Trait for converting an object or actor into the respective ActivityPub type.
+#[async_trait::async_trait(?Send)]
+pub(crate) trait ToApub {
+  type ApubType;
+  async fn to_apub(&self, pool: &DbPool) -> Result<Self::ApubType, LemmyError>;
+  fn to_tombstone(&self) -> Result<Tombstone, LemmyError>;
+}
+
+#[async_trait::async_trait(?Send)]
+pub(crate) trait FromApub {
```

Comment:
Aren't there traits that were used for the forms... those need to be removed if they're unused now.

Reply:
There are now two versions of `FromApub`, one with pub(crate) for `Post` etc, and also [FromApubToForm](https://github.com/LemmyNet/lemmy/pull/1297/files/6cf4e39406b3bce863ec114a7a2355d204facd23#diff-f2f8ae9c33145862dc488df58ddd10c9c96f60d56b5bd6b3e5a828052d108a35R49), which is only used internally by `FromApub`, if parse + db write is needed.